src/dmd/backend/cg87.d

```diff
@@ -3646,6 +3646,21 @@ void fixresult_complex87(ref CodeBuilder cdb,elem *e,regm_t retregs,regm_t *pret
         cdb.genf2(0xDD,modregrm(3,3,0));        // FPOP
         pop87();
     }
+    else if (tym == TYllong)
+    {
+        assert(retregs & mST01, "this float expression is not implemented");
+        assert(*pretregs & mCX, "this float expression is not implemented");
+        pop87();
+        cdb.genfltreg(ESC(MFfloat,1),BX,0);      // FSTP floatreg
+        genfwait(cdb);
+        getregs(cdb,mCX);
+        cdb.genfltreg(LOD, CX, 0);              // MOV EDX,floatreg
+
+        pop87();
+        cdb.genfltreg(ESC(MFfloat,1),BX,0);      // FSTP floatreg
+        genfwait(cdb);
+        cdb.genfltreg(LOD, CX, 0);              // MOV EAX,floatreg
```

Comment:
I guess that cfloat is considered a struct, so it is passed in integer registers. But it fits into RCX, so I'd expect that one of the floats has to be packed into the upper half.
BTW: the comments show other registers.

Reply:
Yeah that's kinda what I gathered from the code.  However, I don't really understand exactly how this code works or how to move the value into the upper part of the register.

It looks like we need to generate code to move the value from retregs to *pretregs.  In this case, retregs is in `ST01`, but I don't even know what that means.  From wikipedia It looks like there are 8 stack registers (`ST1` - `ST7`) but what is `ST01`?  And I don't see where in this code the `ST01` register is being referenced.  I see the comment `floatreg` so maybe that's what `ST01` is?

In any case, this appears to fix the issue, I even added some values to make sure the real part of the float still works.  However, I'm using small values so the upper part may be getting lost.

Reply:
> From wikipedia It looks like there are 8 stack registers (`ST1` - `ST7`) but what is `ST01`?

There are registers ST0-ST7, where ST0 is the one on top of the stack, ST1 is the one below, and so on. mST01 is probably the pair of ST0 and ST1 used for complex numbers.